I2C STM32WL Debug : Unable to link because inline are not inlined

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

-->

### Description of defect

Depending on code, sometimes unable to link using I2C on STM32WL because looks like inline not inlined and thus linker fail to find the function 
<!--
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
-->

#### Target(s) affected by this defect ?

STM32WL (may be other)

#### Toolchain(s) (name and version) displaying this defect ?

![image](https://user-images.githubusercontent.com/2471931/215282545-f7ae24ef-7755-42c2-892e-3296375f3ce3.png)


#### What version of Mbed-os are you using (tag or sha) ?
<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in /platform/mbed_version.h. The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
 
    Master branch is indicated by 'mbed-os-99.99.99
    
    For an issue found on Master please provide the sha being used.
-->

master since  @multiplemonomials commit 0751932dd302c06b0e8e9b95321415a1679d76d0

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)

mbed studio 1.4.4.1

#### How is this defect reproduced ? 

Not really sure sometimes in happens with some code other not. 
I suspect compiler defect under certain circonstances depending on code, in my case target use `mbed-cryptoauthlib` for I2C secure element and another I2C device. in that case it always happen (in `debug` only the one that do not have `--inline` flag on profile)
`develop` and `release`profiles  have no issues

In my case `i2c_is_ready_for_transaction_start()` is not created (not inlined) failed to link (makes sense)

<!--
    For connectivity issues it is recommended to enable mbed trace and attach the traces 
    generated by the logger to the issue. 
-->


```
[Error] @0,0: L6218E: Undefined symbol i2c_is_ready_for_transaction_start (referred from BUILD/EVOLY_T_V12/ARMC6/mbed-os/targets/TARGET_STM/i2c_api.o).
Warning: L3912W: Option 'legacyalign' is deprecated.
Error: L6218E: Undefined symbol i2c_is_ready_for_transaction_start (referred from BUILD/EVOLY_T_V12/ARMC6/mbed-os/targets/TARGET_STM/i2c_api.o).
Finished: 0 information, 1 warning and 1 error messages.
[ERROR] Warning: L3912W: Option 'legacyalign' is deprecated.
Error: L6218E: Undefined symbol i2c_is_ready_for_transaction_start (referred from BUILD/EVOLY_T_V12/ARMC6/mbed-os/targets/TARGET_STM/i2c_api.o).
Finished: 0 information, 1 warning and 1 error messages.
```

anyway changing to always_inline with solve the issue (not sure it's the best option)

```cpp
inline bool i2c_is_ready_for_transaction_start(stm_i2c_state state)
{
    // Note: We can safely send a transaction start in the middle of any single byte operation; this creates a
    // repeated start.

    return state == STM_I2C_IDLE || STM_I2C_PENDING_START == state
           || state == STM_I2C_SB_READ_IN_PROGRESS || state == STM_I2C_SB_WRITE_IN_PROGRESS;
}
```

```cpp
MBED_FORCEINLINE bool i2c_is_ready_for_transaction_start(stm_i2c_state state)
```

--- a/targets/TARGET_STM/i2c_api.c
+++ b/targets/TARGET_STM/i2c_api.c
@@ -893,7 +893,7 @@ int i2c_byte_write(i2c_t *obj, int data)
  * Return whether the given state is a state where we can start a new I2C transaction with the
  * STM32 HAL.
  */
-inline bool i2c_is_ready_for_transaction_start(stm_i2c_state state)
+bool i2c_is_ready_for_transaction_start(stm_i2c_state state)
 {
     // Note: We can safely send a transaction start in the middle of any single byte operation; this creates a
     // repeated start.

